_data-prepper/pipelines/configuration/processors/anomaly-detector.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: default
+title: Anomaly detector
+parent: Processors
+grand_parent: Pipelines
+nav_order: 45
+---
+
+# Anomaly detector
+
+## Overview
+
+The `Anomaly detector` processor takes structured data and runs anomaly detection algorithms on fields you can configure in the data. The data must be either an integer or real number in order for the the anomaly detection algorithm to detect anomalies. We recommend that you deploy the `Aggregate` processor in a pipeline before the `Anomaly detector` processor to achieve the best results.  This is because the `Aggregate` processor aggregates events with same keys onto the same host.
```

Comment:
@dlvenable  @asifsmohammed How does this ensure users will get the best results? How does aggregating events with the same keys onto the same host ensure that those results are better?

Reply:
I do not think we can say "ensure". It is just a recommendation to aggregate events over a time period instead of raw data.  When aggregate processor is used "same keys on to the same host" automatically happens. So, if we are looking for anomaly in latencies from specific IP address(like 1.2.3.4) if all these events go to same host then that host will have more data about events with 1.2.3.4 IP address, more data results in better training of ML algorithm, which should result in better anomaly detection.